### PR TITLE
Allow specifying two different read lengths in Illumina runs

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-2902

### CODE REVIEW

File: pinery-miso/src/main/resources/ca/on/oicr/pinery/lims/miso/queryAllOrderSamples.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@ SELECT o.sequencingOrderId orderId
         ,lib.NAME libraryId 
         ,bc1.sequence barcode 
         ,bc2.sequence barcode_two 
-        ,sp.paired paired 
+        ,sp.readLength = sp.readLength2 paired
```
Comment (callunity) [inline pinery-miso/src/main/resources/ca/on/oicr/pinery/lims/miso/queryAllOrderSamples.sql]: Doesn't this mean that 10X runs will be described as single-end? I don't think that's accurate.

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java [hunk 2]
```diff
@@ -110,16 +110,48 @@ private void processResults(List<NotificationDto> results) {
...
+              }
+              if (Math.abs(params.getReadLength() - illuminaDto.getReadLengths().get(0)) > 1) {
+                return false;
+              }
+              // If no second read is required, make sure none is provided
+              if (params.getReadLength2() == 0) {
+                return illuminaDto.getReadLengths().size() == 1;
+              }
+              // Otherwise, check the second read matches the right length
+              return Math.abs(params.getReadLength2() - illuminaDto.getReadLengths().get(1)) < 2;
```
Comment (callunity) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java]: ``
2019-09-09 10:03:56 ERROR: Failed to save run: 140103_D00353_0031_BC3E4NACXX
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
  at java.util.ArrayList.rangeCheck(ArrayList.java:657)
  at java.util.ArrayList.get(ArrayList.java:433)
  at uk.ac.bbsrc.tgac.miso.webapp.service.RunScannerClient.lambda$processResults$0(RunScannerClient.java:148)
```

Comment (apmasell) [reply miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/service/RunScannerClient.java]: Fixed. I had a condition backwards.

File: pinery-miso/src/main/resources/ca/on/oicr/pinery/lims/miso/queryAllOrderSamples.sql (hunk 1 above)
Comment (apmasell) [reply pinery-miso/src/main/resources/ca/on/oicr/pinery/lims/miso/queryAllOrderSamples.sql]: We define them as single ended right now, so it's consistent with the legacy. I don't know what actually looks at information. CellRanger and BCL2FASTQ look at the bases mask, so I don't know if this matters. I could make it `readLength2 != 0`, which I guess is more accurate, but might cause all the 10X runs to go stale.

Comment (callunity) [reply pinery-miso/src/main/resources/ca/on/oicr/pinery/lims/miso/queryAllOrderSamples.sql]: Half of our runs have `paired = 1` so I think we're already losing on consistency...
`paired` is used to make the `readLength`, which is independent of the `runBasesMask`. It doesn't look like `pinery_ius` even includes the `read_length`, so I don't think it'll be an analysis issue.

Comment (apmasell) [reply pinery-miso/src/main/resources/ca/on/oicr/pinery/lims/miso/queryAllOrderSamples.sql]: Okay, fixed. Even if `pinery_ius` doesn't use it, it still becomes part of the hash in the LIMS key, so we'll just have to see.

**Pull request merged**